### PR TITLE
Added a concept intro page to the documentation

### PR DESCRIPTION
**Pull request by kaikreuzer**
Also-By: Bernard Philippe <bip.philippe@laposte.net>
Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

**Comment by kaikreuzer** [comment]
@bipphilippe: I took part of your openHAB 2 docs and created this PR for ESH, since I believe we should try to keep the documentation about the concepts here.
In order to be able to merge this PR, it would be great if you could [sign the CLA](https://github.com/eclipse/smarthome/blob/master/CONTRIBUTING.md#legal-considerations) for Eclipse.

Another remark: In order to allow other people to edit images in future, we usually also commit the source version of them. To not have to many different tools, we usually suggest to use https://www.draw.io/ for all diagrams.

File: docs/documentation/concepts/index.md [hunk 1]
```diff
@@ -4,34 +4,30 @@ layout: documentation
...
-* Provide a development environment and tools to foster implementations of extensions. The right tooling can support the emergence of further extensions and thus stimulate future contributions to the project.
-* Create a packaging and demo setups. Although the focus is on the framework, it needs to be shown how to package a real solution from it, which can be used as a starting point and for demo purposes.
-Description
-* The Eclipse SmartHome project is a framework that allows building smart home solutions that have a strong focus on heterogeneous environments, i.e. solutions that deal with the integration of different protocols or standards. Its purpose is to provide a uniform access to devices and information and to facilitate different kinds of interactions with them. This framework consists out of a set of OSGi bundles that can be deployed on an OSGi runtime and which defines OSGi services as extension points.
+**Items** represent (fine-grained) functionality that is used by applications - as user interfaces or automation logic) -. Items have a state and they can receive commands.
```
Comment (maggu2810) [inline docs/documentation/concepts/index.md]: I assume the ) and - after logic can be removed.

**Comment by maggu2810** [comment]
Shouldn't we extend the example a little bit?
I think it misses the difference between channel and items.

A thing (actor) controls two output lights that are represented using channels.
To use the channels in an UI or an automation engine you have to link items to the channels of the thing.
We can also link multiple items to a channel.

File: docs/documentation/concepts/index.md (hunk 1 above)
Comment (kaikreuzer) [inline docs/documentation/concepts/index.md]: fixed.

**Comment by kaikreuzer** [comment]
> I think it misses the difference between channel and items.

This is explained above in the concept, not again within the example.

> We can also link multiple items to a channel.

Ok, added a note about that.

Feel free to further enhance the documentation, once this PR is merged.

**Comment by kaikreuzer** [comment]
Oops, I deleted my branch before this PR was merged, so it got automatically closed.

@maggu2810 ok to merge this now?

**Comment by maggu2810** [comment]
Do you want to have this in the 0.8.x branch, too?
I have already created one.
So it would be nice if we can handle #1601 first.
If it should be exist in the 0.8.x branch too, I will merge this one two both branches later.
Ok?

**Comment by kaikreuzer** [comment]
It is enough to have it on master. I just merged #1601.

**Pull request merged**